halpybot/commands/settings.py

```diff
@@ -57,52 +55,17 @@ async def cmd_prefix(ctx: Context, args: List[str]):
         return await ctx.reply(get_help_text("settings prefix"))
     logger.info(
         "PREFIX CHANGE from {prefix} to {new} by {sender}",
-        prefix=config["IRC"]["commandPrefix"],
+        prefix=config.irc.command_prefix,
         new=args[0],
         sender=ctx.sender,
     )
-    config_write("IRC", "commandPrefix", args[0])
+    config.irc.command_prefix = args[0]
     await ctx.reply(f"Changed prefix to '{args[0]}'")
     await ctx.bot.message(
         "#cybers", f"Warning, prefix changed to {args[0]} by {ctx.sender}!"
     )
 
 
-@Settings.command("offline")
-@Require.permission(Cyberseal)
-@Require.channel()
-async def cmd_offline(ctx: Context, args: List[str]):
-    """
-    Change the status of Offline mode.
-
-    Usage: !bot_management offline [Status]
-    Aliases: settings offline
-    """
-    if len(args) == 0:
-        return await ctx.reply(
-            f"{get_help_text('settings offline')}\nCurrent "
-            f"offline setting: {config['Offline Mode']['enabled']}"
-        )
-    if args[0].casefold() == "true" and config["Offline Mode"]["enabled"] != "True":
-        set_to = "True"
-    elif args[0].casefold() == "false" and config["Offline Mode"]["enabled"] != "False":
-        set_to = "False"
-    else:
-        return await ctx.reply(
-            "Error! Invalid parameters given or already in mode. Status not changed."
-        )
-
-    logger.info(
-        "OFFLINE MODE CHANGE from {mode} to {new} by {sender}",
-        mode=config["Offline Mode"]["enabled"],
-        new=set_to.upper(),
-        sender=ctx.sender,
-    )
-    # Write changes to config file
-    config_write("Offline Mode", "enabled", f"{set_to}")
-    await ctx.reply(f"Warning! Offline Mode Status Changed to {set_to.upper()}")
-
-
```

Comment:
While removing the config write part of the offline mode was approved, the ability to change the value of the OLM by command should be maintained.

Reply:
ok. Can i make this function indepotent? I can simplify the logic this way.

Reply:
function restored. I also made it indempotent to make the logic cleaner.